src/log.rs

```diff
@@ -1,458 +0,0 @@
-        .arg(Arg::new("verbose")
-             .short('v')
-             .long("verbose")
-             .multiple_occurrences(true)
-             .help("Log more information, twice or thrice for even more")
-        )
-        .arg(Arg::new("quiet")
-             .short('q')
-             .long("quiet")
-             .multiple_occurrences(true)
-             .conflicts_with("verbose")
-             .help("Log less information, twice for no information")
-        )
-        .arg(Arg::new("syslog")
-             .long("syslog")
-             .help("Log to syslog")
-        )
-        .arg(Arg::new("syslog-facility")
-             .long("syslog-facility")
-             .takes_value(true)
-             .default_value("daemon")
-             .help("Facility to use for syslog logging")
-        )
-        .arg(Arg::new("logfile")
-             .long("logfile")
-             .takes_value(true)
-             .value_name("PATH")
-             .help("Log to this file")
-        )
-    }
-
-    /// Update the logging configuration from command line arguments.
-    ///
-    /// This should be called after the configuration file has been loaded.
-    pub fn update_with_arg_matches(
-        &mut self,
-        matches: &ArgMatches,
-        cur_dir: &Path,
-    ) -> Result<(), Failed> {
-        // log_level
-        for _ in 0..matches.occurrences_of("verbose") {
-            self.log_level.increase()
-        }
-        for _ in 0..matches.occurrences_of("quiet") {
-            self.log_level.decrease()
-        }
-
-        self.apply_log_matches(matches, cur_dir)?;
-
-        Ok(())
-    }
-
-    /// Applies the logging-specific command line arguments to the config.
-    ///
-    /// This is the Unix version that also considers syslog as a valid
-    /// target.
-    #[cfg(unix)]
-    fn apply_log_matches(
-        &mut self,
-        matches: &ArgMatches,
-        cur_dir: &Path,
-    ) -> Result<(), Failed> {
-        if matches.is_present("syslog") {
-            self.log_target = LogTarget::Syslog;
-            if let Some(value) = 
-                Self::from_str_value_of(matches, "syslog-facility")?
-            {
-                self.log_facility = value
-            }
-        }
-        else if let Some(file) = matches.value_of("logfile") {
-            if file == "-" {
-                self.log_target = LogTarget::Stderr
-            }
-            else {
-                self.log_target = LogTarget::File;
-                self.log_file = cur_dir.join(file).into();
-            }
-        }
-        Ok(())
-    }
-
-    /// Applies the logging-specific command line arguments to the config.
-    ///
-    /// This is the non-Unix version that does not use syslog.
-    #[cfg(not(unix))]
-    #[allow(clippy::unnecessary_wraps)]
-    fn apply_log_matches(
-        &mut self,
-        matches: &ArgMatches,
-        cur_dir: &Path,
-    ) -> Result<(), Failed> {
-        if let Some(file) = matches.value_of("logfile") {
-            if file == "-" {
```

Comment:
At a glance I didn't see support for "-" in daemonbase as a file path causing stderr logging to be used instead.

Reply:
And trying `--logfile -` causes a `-` file to be created in my local directory.

Reply:
This should indeed be fixed in _daemonbase._

Reply:
This is addressed in https://github.com/NLnetLabs/daemonbase/pull/1